airflow/operators/python.py

```diff
@@ -178,6 +178,19 @@ class BranchPythonOperator(PythonOperator, SkipMixin):
 
     def execute(self, context: Dict):
         branch = super().execute(context)
+        # TODO: The logic should be moved to SkipMixin to be available to all branch operators.
+        if isinstance(branch, str):
+            branches = {branch}
+        elif isinstance(branch, list):
+            branches = set(branch)
+        else:
+            raise AirflowException("Branch callable must return either a task ID or a list of IDs")
+        valid_task_ids = set(context["dag"].task_ids)
+        invalid_task_ids = branches - valid_task_ids
+        if invalid_task_ids:
+            raise AirflowException(
+                f"Branch callable must valid task_ids. Invalid tasks found: {invalid_task_ids}"
```

Comment:
@eladkal Did you mean something like
`Branch callable must *return* valid task_ids`?

Reply:
Ah yeah. Ill raisr a PR to fix it.

Reply:
https://github.com/apache/airflow/pull/18623

Reply:
LGTM